### PR TITLE
Security fix. (SQL Injection)

### PR DESCRIPTION
**Pull request by jessereitsma**
Issue #1 is fixed with this commit. :+1: 
Thanks @smiegles

### CODE REVIEW

**Comment by smiegles** [comment]
Great, nice to see whe where able to handle this issue so fast for the vendor. I still would recommend @HetIsNiels to read the following article: http://www.dreamhost.com/dreamscape/2013/05/22/php-security-user-validation-and-sanitization-for-the-beginner/

File: src/content/heading.php [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?php
 if(!isset($user))
 {
-	DB::Query("DELETE FROM `users_online` WHERE `ip` = '" . $_SERVER['REMOTE_ADDR'] . "' AND `u_a` = '" . $_SERVER['HTTP_USER_AGENT'] . "'");
-	DB::Query("INSERT INTO `users_online` (user_id, u_a, ip, last_active, last_page) VALUES (0, '" . $_SERVER['HTTP_USER_AGENT'] . "', '" . $_SERVER['REMOTE_ADDR'] . "', CURRENT_TIMESTAMP(), '" . DB::Escape($_SERVER['REQUEST_URI']) . "')");
+	DB::Query("DELETE FROM `users_online` WHERE `ip` = '" . $_SERVER['REMOTE_ADDR'] . "' AND `u_a` = '" . Glybe::Security($_SERVER['HTTP_USER_AGENT']) . "'");
+	DB::Query("INSERT INTO `users_online` (user_id, u_a, ip, last_active, last_page) VALUES (0, '" . Glybe::Security($_SERVER['HTTP_USER_AGENT']) . "', '" . $_SERVER['REMOTE_ADDR'] . "', CURRENT_TIMESTAMP(), '" . DB::Escape($_SERVER['REQUEST_URI']) . "')");
 }
 if(isset($user))
```
Comment (jessereitsma) [inline src/content/heading.php]: Still not getting why you didn't use `else` statement here.. :-1:

**Comment by smiegles** [comment]
Me neither, mind clarifying that?

**Comment by jessereitsma** [comment]
I notice now that there is also a `DB::Escape()` function. 
What's the different?

**Comment by jessereitsma** [comment]
Ah, `Glybe::Security` also adds `htmlspecialchars` which is not necessary in this case.
So i'll change it to `DB::Escape()`

**Comment by smiegles** [comment]
@HetIsNiels, can you tell me where you documented the functions. I can't find the documentation.

**Comment by smiegles** [comment]
Also, would you mind to give comments to the functions. I can't find out what's the exact purpose of every function. More info: http://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.pkg.html

**Comment by RaoulV** [comment]
Why would you expect @HetIsNiels to document the source? As for as I know he isn't currently maintaining this, as the source is published for learning purposes.

**Comment by HetIsNiels** [comment]
The source is public for people to learn. Documentation was never made when this script was made and is not necessary. Also this bug was made by another author. Thanks anyway.

**Comment by jessereitsma** [comment]
@RaoulV well, every project this large should be documented.
What if a new dev joins your dev team? Are u supposing that he will find the whole thing out all by himself? Without documentation? 

Follow the instructions on the following sign :wink:

**Comment by HetIsNiels** [comment]
@jessereitsma if you read my reaction, there is NO current development team on the project.

**Comment by jessereitsma** [comment]
@HetIsNiels `Also this bug was made by another author.` So? You host this.
You can blame another author but you didn't fix it. So blame yourself.

**Comment by HetIsNiels** [comment]
@jessereitsma yeah thanks for fixing your own bug.

**Comment by smiegles** [comment]
image

**Pull request merged**